### PR TITLE
CBAPI-4656 & CBAPI-4567: New stub for Audit Log Request + documentation

### PR DESCRIPTION
**Pull request by abowersox-cb**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Tests have been added that prove the fix is effective or that the feature works.
- [x] New and existing tests pass locally with the changes.
- [x] Code follows the style guidelines of this project (PEP8, clean code, linter).
- [x] A self-review of the code has been done.

## What is the ticket or issue number?

https://jira.carbonblack.local/browse/CBAPI-4656, https://jira.carbonblack.local/browse/CBAPI-4657

## Pull Request Description

Moved Audit Log Requests to a new function entry point under Platform; the actual `AuditLog` class will be expanded upon later once the audit log support in the API is enhanced. Added a Guide doc that shows how it can be used.

Old Audit Log function (under CBCloudAPI) still works, but emits a deprecation warning into the logs.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## How Has This Been Tested?

Unit test for original Audit Log Request has been adapted to new one. Live log of run against `prod02` is here: 
[audit_logs.log](https://github.com/carbonblack/carbon-black-cloud-sdk-python/files/11938940/audit_logs.log)

### CODE REVIEW

File: docs/cbc_sdk.audit_remediation.rst [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Audit and Remediation Package
+cbc\_sdk.audit\_remediation package
```
Comment (avanbrunt-cb) [inline docs/cbc_sdk.audit_remediation.rst:1]: Why are these being reverted?

File: src/tests/unit/platform/test_audit.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+# ==================================== UNIT TESTS BELOW ====================================
+
+
+def test_no_create_object_for_now(cb):
+    """Validates that we can't create an AuditLog object. Remove when we have a better implementation."""
+    with pytest.raises(NotImplementedError):
+        AuditLog(cb, 0)
+
+
+def test_get_auditlogs(cbcsdk_mock):
```
Comment (avanbrunt-cb) [inline src/tests/unit/platform/test_audit.py:44]: Can we add or move an existing test which covers `api.get_auditlogs()` into this file

File: docs/cbc_sdk.audit_remediation.rst (hunk 1 above)
Comment (abowersox-cb) [reply docs/cbc_sdk.audit_remediation.rst:1]: All those files got regenerated when I rebuilt the docs.

Comment (avanbrunt-cb) [reply docs/cbc_sdk.audit_remediation.rst:1]: Please undo this change as we did this intentionally as well as any other package name title changes

File: src/tests/unit/platform/test_audit.py (hunk 2 above)
Comment (abowersox-cb) [reply src/tests/unit/platform/test_audit.py:44]: That test is in `test_rest_api.py` with all the other tests that go with the `CBCloudAPI` object.

File: docs/audit-log.rst [hunk 3]
```diff
@@ -0,0 +1,49 @@
+Audit Log Notifications
```
Comment (avanbrunt-cb) [inline docs/audit-log.rst:1]: I'm hesitant to use the word notifications as that is a different CBC concept and is not used in conjunction with Audit logs in the User Guide. Can we try to scrub all references to notification in the audit log guide/docstrings

File: src/tests/unit/platform/test_audit.py (hunk 2 above)
Comment (avanbrunt-cb) [reply src/tests/unit/platform/test_audit.py:44]: I didn't notice we deprecated it at first resolving

File: docs/audit-log.rst (hunk 3 above)
Comment (abowersox-cb) [reply docs/audit-log.rst:1]: The word "notifications" is used on Dev Network for this API, too: https://developer.carbonblack.com/reference/carbon-black-cloud/cb-defense/latest/rest-api/#audit-log-events

File: docs/cbc_sdk.audit_remediation.rst (hunk 1 above)
Comment (abowersox-cb) [reply docs/cbc_sdk.audit_remediation.rst]: Fixed, but I wish the doc building process didn't step on those titles.

File: docs/audit-log.rst (hunk 3 above)
Comment (abowersox-cb) [reply docs/audit-log.rst]: But I removed the word "notifications" entirely.

Comment (avanbrunt-cb) [reply docs/audit-log.rst]: That is our legacy API documentation and the future is to move away from integrationServices so I'd like to move towards the same terminology used in the User Guide which does not use notifications

**Review by avanbrunt-cb — APPROVED** [review]

**Pull request merged**